packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -518,6 +518,26 @@ describe("Runtime", () => {
                     assert.strictEqual(2, batchEnd, "Did not receive correct batchEnd event for the batch");
                 });
 
+                it("Op send fails while processing another op", async () => {
+                    const clientId: string = "test-client";
+                    const message: Partial<ISequencedDocumentMessage> = {
+                        clientId,
+                        type: MessageType.Operation,
+                    };
+
+                    await deltaManager.inbound.pause();
+                    pushOp(message);
+                    pushOp(message);
+                    pushOp(message);
+                    await processOps();
```

Comment:
I'm not sure what this test actually testing. It does not result in ops being generated while processing an op.
I.e. it will either pass of fail in the same way, with or without your other changes.
Plus, if I got it right, this code uses MockDeltaManager, not real (production) code you are modifying, and there is no ContainerRuntime created in this test (as far as I can see).

I think the easiest way to validate new behavior is to use some existing DDS (in end-to-end tests) and change it from within event handler.